### PR TITLE
aider: init at 0.14.2

### PR DESCRIPTION
**Pull request by nwjsmith**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/aider-chat/default.nix [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+, tiktoken
+, tqdm
+, urllib3
+, wcwidth
+, yarl
+}:
+
+buildPythonPackage rec {
+  pname = "aider-chat";
+  version = "0.14.1";
```
Comment (fabaff) [inline pkgs/development/python-modules/aider-chat/default.nix:46]:
```suggestion
  version = "0.14.1";
  format = "setuptools";
```

File: pkgs/development/python-modules/aider-chat/default.nix [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  ];
+
+  postPatch = ''
+    substituteInPlace aider/repomap.py \
+      --replace '"ctags"' '"${universal-ctags}/bin/ctags"'
+  '';
+
+  # Tests require a Git repository
+  doCheck = false;
+
```
Comment (fabaff) [inline pkgs/development/python-modules/aider-chat/default.nix:101]: As an additional safe-guard.

```suggestion

  pythonImportsCheck = [
    "aider"
  ];

```

File: pkgs/development/python-modules/aider-chat/default.nix [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+, urllib3
+, wcwidth
+, yarl
+}:
+
+buildPythonPackage rec {
+  pname = "aider-chat";
+  version = "0.14.1";
+
+  disabled = pythonOlder "3.8";
```
Comment (fabaff) [inline pkgs/development/python-modules/aider-chat/default.nix:48]: See https://github.com/paul-gauthier/aider/blob/v0.14.1/setup.py#L21

```suggestion
  disabled = pythonOlder "3.9";
```

**Review by fabaff — COMMENTED** [review]
Please move the change for the `maintainers/maintainer-list.nix` file to separate commit in this PR.

**Comment by nwjsmith** [comment]
@fabaff all set, thanks for the review!

**Comment by elohmeier** [comment]
For anyone landing here via search, this is the package I'm currently using: https://github.com/elohmeier/ptsd/blob/master/packages/aider/default.nix
Planning to upstream it into nixpkgs at some point.

**Comment by nwjsmith** [comment]
@elohmeier fantastic! Excited to see this in nixpkgs.

**Comment by tylerjl** [comment]
Awesome @elohmeier - when you do upstream your package, I'll volunteer to be listed as a `maintainer` if you want any help keeping the package up-to-date :+1:

**Comment by taha-yassine** [comment]
I tried packaging it here: https://github.com/NixOS/nixpkgs/pull/323927
If anyone wants to be added as a maintainer lmk

**Pull request closed without merging**